src/js/components/AddressBox.jsx

```diff
@@ -0,0 +1,82 @@
+import React, { Component, PropTypes } from "react";
+import { ButtonToolbar } from "react-bootstrap";
+import VoterStore from "../stores/VoterStore";
+import LoadingWheel from "../components/LoadingWheel";
+import VoterActions from "../actions/VoterActions";
+
+export default class AddressBox extends Component {
+  static propTypes = {
+    history: PropTypes.object,
+    saveUrl: PropTypes.string.isRequired
+  };
+
+  constructor (props) {
+      super(props);
+      this.state = { loading: false };
+  }
+
+  componentDidMount () {
+    this.setState({ location: VoterStore.getAddress() });
+    this.listener = VoterStore.addListener(this._onChange.bind(this));
+  }
+
+  componentWillUnmount (){
+    this.listener.remove();
+  }
+
+  _onChange () {
+    if (this.state.location){
+      this.props.history.push(this.props.saveUrl);
+    } else {
+      this.setState({ location: VoterStore.getAddress(), loading: false });
+    }
+  }
+
+  _ballotLoaded (){
+    console.log("ballotLoaded");
+    this.props.history.push(this.props.saveUrl);
+  }
+
+  updateLocation (e) {
+    this.setState({
+      location: e.target.value
+    });
+  }
+
+  saveLocation (e) {
+    e.preventDefault();
+    var { location } = this.state;
+    console.log("Saving location", location);
+    VoterActions.saveAddress(location);
+    this.setState({loading: true});
+  }
+
+  render () {
+    var { loading, location } = this.state;
```

Comment:
Might want to get more into using let so it gets stored yet will not go up the prototype chain as fast as var will. There is more reliable scoping with let and we will be using a rule in linting where no var will not allow for use of var.